packages/google_sign_in/lib/src/device_flow_widget.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+/// The [GlobalKey] that identifies a [NavigatorState].
+///
+/// This object refers to a navigator used to push the "Device Flow Widget"
+/// that shows "verification url" and "user code" which are
+/// required to authorize sign-in.
+///
+/// The navigatorKey.currentContext is non-null if navigatorKey
+/// is assigned to a valid [Navigator].
+GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+/// Closes the widget that was shown from [showDeviceFlowWidget].
+void closeDeviceFlowWidget() {
+  navigatorKey.currentState!.pop();
+}
+
+/// Displays a widget that shows [code] and [verificationUrl].
+void showDeviceFlowWidget({
+  required String code,
+  required Uri verificationUrl,
+  required Duration expiresIn,
+  VoidCallback? onExpired,
+  VoidCallback? onCanceled,
+}) {
+  showDialog<void>(
+    context: navigatorKey.currentContext!,
+    barrierDismissible: false,
+    builder: (BuildContext context) {
+      return AlertDialog(
+        title: const Text('Google SignIn'),
+        content: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            const Text(
+              'From a PC, phone, or tablet go to:',
```

Comment:
nit: Remove the trailing comma for consistency.

Reply:
Fixed